x-pack/plugins/observability_solution/dataset_quality/public/components/dataset_quality_details/details/dataset_summary.tsx

```diff
@@ -39,12 +40,19 @@ export function DatasetSummary() {
     ? dataFormatter.convert(dataStreamSettings.createdOn)
     : '-';
 
-  return (
+  return !integrationDetailsLoaded ? (
+    <EuiSkeletonRectangle
+      width="100%"
+      height="200px"
+      data-test-subj="datasetQualityDetailsDetailsSectionLoading"
+      className="datasetQualityDetailsDetailsSectionLoading"
+    />
+  ) : (
     <EuiPanel hasBorder={false} hasShadow={false} paddingSize="none">
       <Fragment>
         <FieldsList
           fields={[
-            ...(integrationDetails?.integration
+            ...(integrationDetails?.integration?.integration
```

Comment:
I'm not sure what `integrationDetails.integration.integration`  represents but I guess is an object, although it's not clear what the first 2 accesses are, some renaming might help for future debugging and code maintenance.

Reply:
I too thought the same but due to lack of time was not 100% convinced to do the change.

`IntegrationDetails` define 2 items - response from `Integration API` and response from `integration dashboard API`.

So integrationDetails.integration here means response from Integration API.

Now as part of this PR, this API response changed and it returns `isIntegration`, `areAssetsAvailable` and the actual `integration` object. Hence this long name